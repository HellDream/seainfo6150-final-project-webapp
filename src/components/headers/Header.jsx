import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import styles from './Header.module.css';
import Dialog from '../dialog/Dialog.jsx';
const Header = (props) => {
    const [openDialog, setOpenDialog] = useState(false);
    const [message, setMessage] = useState('');

    const onCancel = (event) => {
        setMessage("You haven't finished yet. Are you sure to cancel?");
        setOpenDialog(!openDialog);
    };

    const backBtn = props.back ? (
        <button className={styles.backBtn} onClick={props.back}></button>
    ) : (
        <div className={styles.empty}></div>
    );

    return (
        <div>
            <Dialog
                open={openDialog}
                onClose={setOpenDialog}
                message={message}
                ok={props.onCancel}
            />

            <header className={styles.header}>
                <div className={styles.navbar}>
                    {backBtn}
                    <a className={styles.homeLink} href="/">
                        <h3 className={styles.title}>thrift store</h3>
                    </a>
                    <div className={styles.links}>
                        {!props.createPage && (
                            <a className={styles.createLink} href="/create">
                                create
                            </a>
                        )}

                        {props.createPage && (
                            <button
                                className={styles.cancel}
                                onClick={onCancel}
                            >
                                cancel
                            </button>
                        )}
                    </div>
                </div>
            </header>
        </div>
    );
};

export default Header;
