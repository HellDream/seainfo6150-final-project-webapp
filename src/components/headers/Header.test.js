import React from 'react';
import Header from './Header';
import { render } from '@testing-library/react'

describe('Header tests', () => {
    it('renders correctly', () => {
        const { container } = render(<Header />);
        expect(container).toMatchSnapshot();
    });
});
  