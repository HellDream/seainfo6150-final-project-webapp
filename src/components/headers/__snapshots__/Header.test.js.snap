// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests renders correctly 1`] = `
<div>
  <div>
    <div />
    <header
      class="header"
    >
      <div
        class="navbar"
      >
        <div
          class="empty"
        />
        <a
          class="homeLink"
          to="/"
        >
          <h3
            class="title"
          >
            thrift store
          </h3>
        </a>
        <div
          class="links"
        >
          <a
            class="createLink"
            to="/create"
          >
            create
          </a>
        </div>
      </div>
    </header>
  </div>
</div>
`;
